### ISSUE TITLE
Missing "Assigned but not used" warning when same symbol used in local function

### ISSUE THREAD

**Issue by fredrikekre** [julia-vscode/StaticLint.jl#314]
This doesn't warn for `Assigned but not used` for `k`. (Perhaps it should also warn for `bar`?)
```
function foo()
    bar(k) = k + 1
    k = 1
    return nothing
end
```

**Comment by pfitzseb** [comment]
Same as https://github.com/julia-vscode/StaticLint.jl/issues/313.

**Comment by fredrikekre** [comment]
Same underlying issue you mean? Cause #313 warns when it shouldn't, but this doesn't warn.

**Comment by pfitzseb** [comment]
Yea.
